src/mapnode.cpp

```diff
@@ -603,8 +603,8 @@ u8 MapNode::getLevel(const NodeDefManager *nodemgr) const
 		if (level)
 			return level;
 	}
-	if (f.leveled > LEVELED_MAX)
-		return LEVELED_MAX;
+	if (f.leveled > f.leveled_max)
+		return f.leveled_max;
```

Comment:
Please add `// Return static value from nodedef if param2 isn't used for level`
This confused me at first

You can also replace the if + returns with `return MYMIN(f.leveled, f.leveled_max);`